homeassistant/components/fireservicerota/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Binary Sensor platform for FireServiceRota integration."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN as FIRESERVICEROTA_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up FireServiceRota binary sensor based on a config entry."""
+    coordinator = hass.data[FIRESERVICEROTA_DOMAIN][entry.entry_id]
+
+    async_add_entities([ResponseBinarySensor(coordinator, entry)], True)
+
+
+class ResponseBinarySensor(BinarySensorEntity, CoordinatorEntity):
+    """Representation of an FireServiceRota sensor."""
+
+    def __init__(self, coordinator: DataUpdateCoordinator, entry):
+        """Initialize."""
+        self._coordinator = coordinator
+        super().__init__(coordinator)
+        self._unique_id = entry.unique_id
+
+        self._state = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return "Duty"
+
+    @property
+    def icon(self) -> str:
+        """Return the icon to use in the frontend."""
+        return "mdi:calendar"
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this binary sensor."""
+        return f"{self._unique_id}_Duty"
+
+    @property
+    def is_on(self) -> str:
+        """Return the status of the binary sensor."""
+        return self._state
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed."""
+        return False
+
+    @property
+    def state(self) -> str:
```

Comment:
Binary sensor entities should not override `state`. Only implement `is_on` instead.

Reply:
Nice, done.

Reply:
Hmm, with only is_on() and no state() the state doesn't get updated from on to off, once the schedule changes, the attributes are updated.. can this be a coordinator<>binary_sensor issue?

Reply:
If I keep the state() method, all works fine, it switches back and forth every minute when schedule has changed, and if I add True to device init it all works like it should.. weird.

Reply:
If that's the case, make sure you update the coordinator inside `async_setup_entry` in `__init__.py` before forwarding the config entry.

Reply:
I call async_availability_update() as one of the first things after init of the coordinator in async_setup_entry()
Need to add some debug logging to see whats happening. Also with only is_on() the binary_sensor state never changes to the correct one.

Reply:
`async_availability_update` returns the value, it doesn't store it on the coordinator. 

Please read the documentation about https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

All other methods on an UpdateCoordinator don't belong there and when used, make it easy to cause issues.

Reply:
My coordinator wasn't just a coordinator, after looking at some other integrations with DataUpdateCoordinators and the development guide I moved the coordinator code to async_setup_entry and renamed 'my coordinator' to 'client' this fixed the initial fetching of data for binary_sensor!
I hope you agree with this setup, I changed more than I wanted at first,  but it looks more logical to me and works like it should
One strange this is still there; if I return True and False inside is_on() al works 100% reliable, the state of the binary sensor is correctly toggled to on/off in the GUI when schedule data is fetched and updated. But when I return STATE_ON and STATE_OFF, the state stays on On in the GUI, even when debug log inside is_on() says it changed the state to 'off'! looks like a bug to me?! The attribute data is changing correctly regardless, this proofs that self.coordinator.data is correctly updated as well.

Reply:
Silly me, is_on() should return a boolean, so the erratic behavior was to be expected, I copied the code from state() over without converting to boolean. It's fine now and works like it should.
And without the forced (and double) update on setup. @balloob please re-review if you have some spare time. Thanks!

Reply:
If only 1 entity consumes it, you don't even need an update coordinator. You could just put that in `async_update` of the entity too. This is fine though.